crates/kornia-linalg/src/linalg.rs

```diff
@@ -0,0 +1,466 @@
+
+impl IndexedQuat {
+    fn new(q: Quat) -> Self {
+        IndexedQuat(q)
+    }
+
+    fn to_quat(&self) -> Quat {
+        self.0
+    }
+}
+
+impl Index<usize> for IndexedQuat {
+    type Output = f32;
+
+    fn index(&self, index: usize) -> &Self::Output {
+        match index {
+            0 => &self.0.x,
+            1 => &self.0.y,
+            2 => &self.0.z,
+            3 => &self.0.w,
+            _ => panic!("Index out of bounds for Quaternion: {}", index),
+        }
+    }
+}
+
+impl IndexMut<usize> for IndexedQuat {
+    fn index_mut(&mut self, index: usize) -> &mut Self::Output {
+        match index {
+            0 => &mut self.0.x,
+            1 => &mut self.0.y,
+            2 => &mut self.0.z,
+            3 => &mut self.0.w,
+            _ => panic!("Index out of bounds for Quaternion: {}", index),
+        }
+    }
+}
+
+/// Function used to apply a givens rotation S. Calculates the weights and updates the quaternion to contain the cumultative rotation
+fn jacobi_conjugation(x: usize, y: usize, z: usize, S: &mut Symmetric3x3, q: &mut IndexedQuat) {
+    // Compute the Givens rotation (approximated)
+    let mut g = approximate_givens_quaternion(S);
+    // Scale and calculate intermediate values
+    let scale = 1.0 / (g.ch * g.ch + g.sh * g.sh);
+    let a = (g.ch * g.ch - g.sh * g.sh) * scale;
+    let b = 2.0 * g.sh * g.ch * scale;
+
+    // Create a copy of the matrix to avoid modifying the original during calculations
+    let mut _S = S.clone();
+
+    // Perform conjugation: S = Q'*S*Q
+    S.m_00 = a * (a * _S.m_00 + b * _S.m_10) + b * (a * _S.m_10 + b * _S.m_11);
+    S.m_10 = a * (-b * _S.m_00 + a * _S.m_10) + b * (-b * _S.m_10 + a * _S.m_11);
+    S.m_11 = -b * (-b * _S.m_00 + a * _S.m_10) + a * (-b * _S.m_10 + a * _S.m_11);
+    S.m_20 = a * _S.m_20 + b * _S.m_21;
+    S.m_21 = -b * _S.m_20 + a * _S.m_21;
+    S.m_22 = _S.m_22;
+
+    // Update cumulative rotation qV
+    let mut tmp = [0.0, 0.0, 0.0];
+    tmp[0] = q[0] * g.sh;
+    tmp[1] = q[1] * g.sh;
+    tmp[2] = q[2] * g.sh;
+    g.sh *= q[3];
+
+    // (x, y, z) corresponds to (0,1,2), (1,2,0), (2,0,1) for (p, q) = (0,1), (1,2), (0,2)
+    q[z] = q[z] * g.ch + g.sh;
+    q[3] = q[3] * g.ch - tmp[z]; // w
+    q[x] = q[x] * g.ch + tmp[y];
+    q[y] = q[y] * g.ch - tmp[x];
+
+    // Re-arrange matrix for next iteration
+    _S.m_00 = S.m_11;
+    _S.m_10 = S.m_21;
+    _S.m_11 = S.m_22;
+    _S.m_20 = S.m_10;
+    _S.m_21 = S.m_20;
+    _S.m_22 = S.m_00;
+
+    S.m_00 = _S.m_00;
+    S.m_10 = _S.m_10;
+    S.m_11 = _S.m_11;
+    S.m_20 = _S.m_20;
+    S.m_21 = _S.m_21;
+    S.m_22 = _S.m_22;
+}
+
+/// Function used to contain the givens permutations and the loop of the jacobi steps controlled by JACOBI_STEPS
+/// Returns the quaternion q containing the cumultative result used to reconstruct S
+fn jacobi_eigenanalysis(mut S: Symmetric3x3) -> Mat3 {
+    let mut q = IndexedQuat::new(Quat::from_xyzw(0.0, 0.0, 0.0, 1.0));
+    for _i in 0..JACOBI_STEPS {
```

Comment:
Maybe play with the num steps to reduce and find balance between speed and results quality

Reply:
That worked changing JACOBI_STEPS from 12 to 6 is accurate and more speed than faer:
![Code_itWQQcKALL](https://github.com/user-attachments/assets/8eb450c4-1af6-4647-8a19-3c6b251094ab)

Reply:
amazing! can we make sure with tests that the function produce proper results under some corner cases ? test with a couple different input matrices. Good job!